### ISSUE TITLE
False positive with __slots__, inheritance and C extensions

### ISSUE THREAD

**Issue by Juanlu001** [PyCQA/pylint#2807]
### Steps to reproduce

```
$ cat test.py
import ephem

o = ephem.Observer()
o.lon = 1
print(o.lon)
$ python test.py
57:17:44.8
```

([Source code of `ephem.Observer`](https://github.com/brandon-rhodes/pyephem/blob/f9c11c2f052765d903f229724132a7a7c4ffdba3/ephem/__init__.py#L255-L286))

### Current behavior

```
$ pylint -E test.py
************* Module test
test.py:4:0: E0237: Assigning to attribute 'lon' not defined in class slots (assigning-non-slot)
```

### Expected behavior

```
$ pylint -E test.py
$
```

### pylint --version output

```
$ pylint --version
pylint 2.3.1
astroid 2.2.5
Python 3.7.1 | packaged by conda-forge | (default, Feb 26 2019, 04:48:14) 
[GCC 7.3.0]
```

**Comment by PCManticore** [comment]
Thanks for reporting the issue @Juanlu001 This is fixed in `master` and it's going to be part of 2.4 when that gets released.

**Comment by Juanlu001** [comment]
Thanks a lot @PCManticore!
